exporters/otlp/otlpmetric/otlpmetricgrpc/internal/oconf/options.go

```diff
@@ -279,6 +280,18 @@ func WithEndpoint(endpoint string) GenericOption {
 	})
 }
 
+func WithEndpointURL(u *url.URL) GenericOption {
```

Comment:
I think that we should accept a `string` and not https://pkg.go.dev/net/url#URL?

> A URL represents a parsed URL (technically, a URI reference).

I see `url.URL` more like a return type of `url.Parse` rather than a type that should be used as input.

Take notice that most of the API of https://pkg.go.dev/net/http accept url as a string.

Also only some fields of `url.URL` are used. E.g what if the user sets only the `RawPath` field of `url.URL`?

Reply:
I did hesitate with that attribute type. I've changed it to a string.